source/fundamentals/builders/aggregates.txt

```diff
@@ -898,3 +905,39 @@ aggregation stage:
 Learn more about this helper from the
 `searchMeta() API documentation <{+api+}/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html#searchMeta(com.mongodb.client.model.search.SearchCollector)>`__.
 
+.. _kotlin-atlas-vector-search:
+
+Atlas Vector Search
+-------------------
+
+.. important::
+
+   To learn about which versions of MongoDB Atlas support this feature, see
+   :atlas:`Limitations </atlas-vector-search/vector-search-stage/#limitations>`
+   in the Atlas documentation.
+
+Use the ``vectorSearch()`` method to create a :atlas:`$vectorSearch </atlas-vector-search/vector-search-stage/>`
+pipeline stage that specifies a **semantic search**. A semantic search is
+a type of search that locates pieces of information that are similar in meaning.
+
+To use this feature, you must set up a vector search index and index your
+vector embeddings. To learn how to set up search indexes in MongoDB Atlas, see
+:atlas:`How to Index Vector Embeddings for Vector Search
+</atlas-search/field-types/knn-vector/>` in the Atlas documentation.
+
+The example in this section uses data modeled with the following Kotlin data class:
+
+.. literalinclude:: /examples/generated/AggregatesBuilderTest.snippet.vector-search-data-class.kt
+   :language: kotlin
+
+The following example shows how to build an aggregation pipeline that uses the
+``vectorSearch()`` and ``project()`` methods to compute a vector search
```

Comment:
Question: Do we need to show the `project()` stage in this example? I think it might be more clear to adjust the snippet to only show the `Aggregates.vectorSearch` stage so it doesn't seem like `project` is always needed (unless it is always needed). Removing project would also help make this sentence a little shorter and easier to read.

Reply:
Sure, I can adjust the example. I can also explain more what the agg stage is actually doing.